iis-core/src/test/java/eu/dnetlib/iis/core/SshOozieClient.java

```diff
@@ -0,0 +1,133 @@
+package eu.dnetlib.iis.core;
+
+import java.io.IOException;
+import java.util.Properties;
+
+import org.apache.oozie.client.WorkflowJob.Status;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Preconditions;
+
+import net.schmizz.sshj.SSHClient;
+
+/**
+ * Service for retrieving informations about jobs on running oozie instance.
+ * It communicates indirectly with oozie through some remote host using ssh protocol.
+ * 
+ * @author madryk
+ *
+ */
+public class SshOozieClient {
+	
+	private Logger log = LoggerFactory.getLogger(getClass());
+	
+	
+	private OozieCmdLineAnswerParser oozieCmdLineParser = new OozieCmdLineAnswerParser();
+	
+	
+	private String remoteHost;
+	
+	private String remoteUser;
+	
+	private String oozieUrl;
+	
+	
+	private SSHClient sshClient;
+	
+	
+	//------------------------ CONSTRUCTORS --------------------------
+	
+	/**
+	 * Default constructor
+	 * 
+	 * @param remoteHost - address of host that will be responsible for communication with oozie
+	 * @param remoteUser - name of user on host machine
+	 * @param oozieUrl - address of oozie (from remote host viewpoint) 
+	 */
+	public SshOozieClient(String remoteHost, String remoteUser, String oozieUrl) {
+		this.remoteHost = remoteHost;
+		this.remoteUser = remoteUser;
+		this.oozieUrl = oozieUrl;
+	}
+	
+	
+	//------------------------ LOGIC --------------------------
+	
+	/**
+	 * Opens ssh connection.<br/>
+	 * Method must be executed before any attempt to read from remote host.
+	 */
+	public void openConnection() throws IOException {
+		if (sshClient != null) {
+			log.warn("Attempt to open new connection when the old one is still opened. Will use old connection.");
+			return;
+		}
```

Comment:
In my opinion it would be better to throw and exception if one wants to connect second time (without first closing the connection).